release/Magarena/scripts/Ajani__Valiant_Protector.groovy

```diff
@@ -12,15 +12,16 @@
         @Override
         public void executeEvent(final MagicGame game, final MagicEvent event) {
             final MagicCardList library = event.getPlayer().getLibrary();
-            def nonTarget = library.takeWhile({ !it.hasType(MagicType.Creature) });
-            if (library.any({ it.hasType(MagicType.Creature) })) {
-                final MagicCard target = library.find({ it.hasType(MagicType.Creature) });
+            def predicate = { final MagicCard card -> card.hasType(MagicType.Creature) };
+            final MagicCardList nonTarget = (MagicCardList)library.takeWhile({ !predicate(it) });
```

Comment:
This doesn't seem work, did you test it in a game? takeWhile returns a List but List can't be cast to MagicCardList, it would be a class cast exception.

Reply:
My bad, seems that Groovy works differently from Java, it calls the conversion constructor automatically so it does work. It will be better to call the conversion constructor directly to reduce the chance of runtime class cast exceptions. i.e. `new MagiCardList(library.takeWhile...)`

Reply:
Generally in Java code we try to avoid casts, so I think we should do the same in Groovy code, although Groovy does some automatic conversions. It will be easier if we keep to the same conventions in both languages.

Reply:
I see. Changed in both Ajani and Tezzeret.